File: Frontend-POS/RBMS-POS-Mobile-Web/src/app/core/api/models/payment-response-model.ts
/* tslint:disable */
/* eslint-disable */
export interface PaymentResponseModel {
  amountDue?: number;
  amountReceived?: number;
  billNumber?: string | null;
  cashierSessionId?: number;
  changeAmount?: number;
  grandTotal?: number;
  guestCount?: number;
  guestType?: string | null;
  note?: string | null;
  orderBillId?: number;
  orderId?: number;
  orderNumber?: string | null;
  paidAt?: string;
  paymentId?: number;
  paymentMethod?: string | null;
  paymentReference?: string | null;
  slipImageFileId?: number | null;
  slipOcrAmount?: number | null;
  slipVerificationStatus?: string | null;
  tableId?: number | null;
  tableName?: string | null;
  zoneName?: string | null;
}

File: Backend-POS/POS.Main/POS.Main.Business.Payment/Models/Payment/PaymentResponseModel.cs
namespace POS.Main.Business.Payment.Models.Payment;

public class PaymentResponseModel
{
    public int PaymentId { get; set; }
    public int OrderBillId { get; set; }
    public int CashierSessionId { get; set; }
    public string PaymentMethod { get; set; } = string.Empty;
    public decimal AmountDue { get; set; }
    public decimal AmountReceived { get; set; }
    public decimal ChangeAmount { get; set; }
    public int? SlipImageFileId { get; set; }
    public decimal? SlipOcrAmount { get; set; }
    public string SlipVerificationStatus { get; set; } = string.Empty;
    public string? PaymentReference { get; set; }
    public DateTime PaidAt { get; set; }
    public string? Note { get; set; }

    // Bill info
    public string? BillNumber { get; set; }
    public decimal GrandTotal { get; set; }

    // Order info
    public int OrderId { get; set; }
    public string? OrderNumber { get; set; }

    // Table info
    public int? TableId { get; set; }
    public string? TableName { get; set; }
    public string? ZoneName { get; set; }

    // Guest info
    public string? GuestType { get; set; }
    public int GuestCount { get; set; }
}
